app.go

```diff
@@ -0,0 +1,234 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	cmn "github.com/tendermint/tendermint/libs/common"
+	dbm "github.com/tendermint/tendermint/libs/db"
+)
+
+const (
+	appName = "ethereum-bridge"
+)
+
+type ethereumBridgeApp struct {
+	*bam.BaseApp
+	cdc *codec.Codec
+
+	keyMain          *sdk.KVStoreKey
+	keyAccount       *sdk.KVStoreKey
+	keyStaking       *sdk.KVStoreKey
+	tkeyStaking      *sdk.TransientStoreKey
+	keyOracle        *sdk.KVStoreKey
+	keyFeeCollection *sdk.KVStoreKey
+	keyParams        *sdk.KVStoreKey
+	tkeyParams       *sdk.TransientStoreKey
+
+	accountKeeper       auth.AccountKeeper
+	bankKeeper          bank.Keeper
+	feeCollectionKeeper auth.FeeCollectionKeeper
+	stakingKeeper       staking.Keeper
+
+	paramsKeeper params.Keeper
+	oracleKeeper oracle.Keeper
+}
+
+// NewEthereumBridgeApp is a constructor function for ethereumBridgeApp
+func NewEthereumBridgeApp(logger log.Logger, db dbm.DB) *ethereumBridgeApp {
+
+	// First define the top level codec that will be shared by the different modules
+	cdc := MakeCodec()
+
+	// BaseApp handles interactions with Tendermint through the ABCI protocol
+	bApp := bam.NewBaseApp(appName, logger, db, auth.DefaultTxDecoder(cdc))
+
+	// Here you initialize your application with the store keys it requires
+	var app = &ethereumBridgeApp{
+		BaseApp: bApp,
+		cdc:     cdc,
+
+		keyMain:          sdk.NewKVStoreKey(bam.MainStoreKey),
+		keyAccount:       sdk.NewKVStoreKey(auth.StoreKey),
+		keyStaking:       sdk.NewKVStoreKey(staking.StoreKey),
+		tkeyStaking:      sdk.NewTransientStoreKey(staking.TStoreKey),
+		keyOracle:        sdk.NewKVStoreKey(oracle.StoreKey),
+		keyFeeCollection: sdk.NewKVStoreKey(auth.FeeStoreKey),
+		keyParams:        sdk.NewKVStoreKey(params.StoreKey),
+		tkeyParams:       sdk.NewTransientStoreKey(params.TStoreKey),
+	}
+
+	// The ParamsKeeper handles parameter storage for the application
+	app.paramsKeeper = params.NewKeeper(app.cdc, app.keyParams, app.tkeyParams)
+
+	// The AccountKeeper handles address -> account lookups
+	app.accountKeeper = auth.NewAccountKeeper(
+		app.cdc,
+		app.keyAccount,
+		app.paramsKeeper.Subspace(auth.DefaultParamspace),
+		auth.ProtoBaseAccount,
+	)
+
+	// The BankKeeper allows you perform sdk.Coins interactions
+	app.bankKeeper = bank.NewBaseKeeper(
+		app.accountKeeper,
+		app.paramsKeeper.Subspace(bank.DefaultParamspace),
+		bank.DefaultCodespace,
+	)
+
+	// The FeeCollectionKeeper collects transaction fees and renders them to the fee distribution module
+	app.feeCollectionKeeper = auth.NewFeeCollectionKeeper(cdc, app.keyFeeCollection)
+
+	app.stakingKeeper = staking.NewKeeper(
+		app.cdc,
+		app.keyStaking, app.tkeyStaking,
+		app.bankKeeper, app.paramsKeeper.Subspace(staking.DefaultParamspace),
+		staking.DefaultCodespace,
+	)
+
+	// The OracleKeeper is the Keeper from the oracle module
+	// It handles interactions with the oracle store
+	oracleKeeper, oracleErr := oracle.NewKeeper(
```

Comment:
this should panic instead of returning an error

Reply:
fixed: https://github.com/musnit/peggy/commit/586e80686df398a42e305fc14b58f0ccacb81697

Reply:
I think this should panic in `NewKeeper` rather than calling panic on the returned error

Reply:
fixing in next push